### ISSUE TITLE
Assignment of item at index in repeated field

### ISSUE THREAD

**Issue by BenRKarl** [googleapis/proto-plus-python#188]
With a normal Python list we can do:

```py
l = [1, 2, 3, 4, 5]
l[1] = 'hello'
print(l)
> [1, 'hello', 3, 4, 5]
```

However for a repeated field of a message wrapped by proto-plus we see slightly different behavior:

```py
l = [1, 2, 3, 4, 5]
l[1] = 'hello'
print(l)
> [1, 'hello', 2, 3, 4, 5]
```

Notice how the `2` just gets push back an index instead of overridden.

For a raw protobuf message, we would normally do:

```py
l = [1, 2, 3, 4, 5]
l[1].CopyFrom('hello')
print(l)
> [1, 'hello', 3, 4, 5]
```

**Comment by software-dov** [comment]
To clarify, this bug seems to indicate that element assignment in a repeated field is incorrectly being interpreted as insertion.

**Comment by BenRKarl** [comment]
Yes exactly.

**Comment by software-dov** [comment]
I took a look through the unit tests, and it seems that this exact situation is tested for (with the correct behavior expected). I think this is a somewhat recent addition by a third party contributor, thought. I will get the change packaged and out soon and see if that solves the problem.

**Comment by BenRKarl** [comment]
Interesting - can you point me to where the unit tests are? Maybe we have a corner case that we can add a test for.

**Comment by 0x2b3bfa0** [comment]
Possible duplicate of #177, allegedly fixed through #178. Salutations from the third party contributor. 🤣

**Comment by BenRKarl** [comment]
Thanks @0x2b3bfa0 #178 does indeed appear to fix our issue.

@software-dov how soon do you think until 1.13.1 will get published to PyPi?
